Add S curve example

frmdstryr opened the pull request:
Please review and comment.  The pins and default speed/accel are changed slightly.

It's based on the equations from https://www.parkermotion.com/manuals/6k/6K_PG_RevB_Chp5_CustomProfiling.pdf 

Thanks!

luni64 commented:
Thanks a lot. I'm currently work on a major redesign of the library. During that I changed the interface you are using. Basically in the old version movements were relative and positive (direction was handled by dir pin only) To get more flexibility I changed that to absolute. I.e. currentPosition, and the other parameters are signed now and represent the absolute position. So, your code will need some adjustments to work with the new interface. 
I'd like to finish the restructuring (very soon)  before merging in (and probably adapt) your code. Would be glad of course if you could do the adaption.

frmdstryr commented:
> To get more flexibility I changed that to absolute

Interesting, good to hear this.  I rewrote teensystep locally just to get a better understanding of how it works and my rewrite does this as well.  

I can share the code if you're interested.

luni64 commented:
Updated the new code to the develop_newRotate branch. https://github.com/luni64/TeensyStep/issues/33#issue-393775187
Would be great if you could adapt your code to the new API

Note: the new Rotational controller is currently not functional.
